Features: Day Right Click, Show Bg Event Title

camargodanny opened the pull request:
**Day Right Click**
New event handler to be able to react when the user right-clicks on any empty cell: a date in month view or time slot in agenda view.

**Show Bg Event Title**
Custom property for the calendar to be able to add some kind of title or description to background events.

Both features are missing in the core component.

PS: Updated the reference to the resources project to use the existing "fullcalendar_resources" which contains sample metadata for events and resources.

camargodanny committed:
Update reference to resources project

camargodanny committed:
- Implement "onDayRightClickMethodID" event handler, new event when right-clicking on an empty date or time in a calendar view
- Implement custom property "showBgEventTitle", used to add some description to background events

paronne commented on fullcalendarcomponent/fullcalendar/fullcalendar.js:
```diff
@@ -336,6 +354,18 @@ angular.module('svyFullcalendar', ['servoy']).directive('svyFullcalendar', funct
+					// fc-bg-title is a custom css class just to differentiate from fc-title
+					// FIXME is this the best way to do it? 
+					if (event.rendering == 'background' && event.title && $scope.model.calendarOptions.showBgEventTitle) {
+						element.append('<div class="fc-bg-title"><span>' + event.title + '</span></div>');
```
What happens when the text is too long ?
Do we need some styling to prevent overflow ?

camargodanny replied:
The background event will be render as the regular events, the text is wrapped.
I don't think it is necessary to implement any styling

paronne commented on fullcalendarComponent$demo/.project:
```diff
@@ -3,12 +3,11 @@
 	<projects>
-		<project>resources</project>
-		<project>fullcalendar_localhost_resources</project>
+		<project>fullcalendar_resources</project>
```
Resource project fullcalendar_resources is indeed the one to be used in sample demo of fullcalendar, however the fullcalendar is used also by sampleGallery (demo.servoy.com) and it needs to share the same default "resources" project

camargodanny replied:
How to have two different resources projects?

camargodanny committed:
- Improve handling of day right-click event, it supports newer versions of the core fullcalendar library and pass the resource to the event handler
- Improve rendering of background events to control whether  they
- Improve view rendering to include all-day slots and non-business hours in Agenda View and prevent issues in Month View
- Add missing resource parameter to the day click event handler

camargodanny committed:
Calculate default date based on the events in the database, current date if nothing is found

camargodanny commented on fullcalendarComponent$demo/forms/demoCalendar.js:
```diff
@@ -213,7 +227,7 @@ function onLoad(event) {
 			},
 			dayNamesShort : dayNamesShort,
-			defaultDate : new Date(2016, 3, 27),
+			defaultDate : new Date(),
```
Implemented function to calculate the default date to use based on the data in the database

camargodanny commented on fullcalendarcomponent/fullcalendar/fullcalendar.js:
```diff
@@ -356,6 +386,25 @@ angular.module('svyFullcalendar', ['servoy']).directive('svyFullcalendar', funct
+						}
+
+						if (targetElements) {
+							targetElements.bind('contextmenu', { event: createMouseEvent(true) }, dayRightClick);
```
Done

camargodanny commented on fullcalendarcomponent/fullcalendar/fullcalendar.js:
```diff
@@ -356,6 +386,25 @@ angular.module('svyFullcalendar', ['servoy']).directive('svyFullcalendar', funct
+						// Agenda View
+						if (element.hasClass('fc-agenda-view')) {
+							// Find all time cells excluding the ones that show the time
+							targetElements = element.find('.fc-slats .fc-widget-content:not(.fc-time)');
```
Added support for all-day events and non-business hours

camargodanny commented on fullcalendarcomponent/fullcalendar/fullcalendar.js:
```diff
@@ -248,6 +249,23 @@ angular.module('svyFullcalendar', ['servoy']).directive('svyFullcalendar', funct
+						var viewObject = $scope.model.view;
+
+						if (cell) {
+							$scope.handlers.onDayRightClickMethodID(parseMoment(cell.start), jsEvent, viewObject);
```
I think it is related to moment UTC, I get time difference only in this demo, in my sample solution it is working fine
